app/controllers/concerns/project_concern.rb

```diff
@@ -0,0 +1,13 @@
+module ProjectConcern
+  extend ActiveSupport::Concern
+
+  private
+
+  def change_date(project)
```

Comment:
I thought inheritance was the answer so we could have a single project-type-agnostic route for change_date, as you suggested above. I think that would remove the need for the `urlComponent` param. There's no reason functions like change_date need to know the project type—they should function the same either way.